filtering/proto.go

```diff
@@ -0,0 +1,148 @@
+package filtering
+
+import (
+	"strings"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/dynamicpb"
+)
+
+type FilterOption func(opts *filterOptions)
+
+type filterOptions struct {
+	filterableFields []string
+}
+
+// WithFilterableFields marks the given fields as filterable.
+// To mark a simple field (native types) or a message field (with all its underlying fields, recursively)
+// as filterable, use the field name.
+// To mark a specific nested field as filterable, use the field name with a dot prefix.
+// For example:
+//   - WithFilterableFields("string_field") marks the string_field as filterable.
+//   - WithFilterableFields("nested_message") marks the nested_message field and all its underlying fields as
+//     filterable.
+//   - WithFilterableFields("nested_message.nested_string") marks the nested_string field as filterable.
+//   - WithFilterableFields("nested_message.nested_string", "nested_message.nested_int32") marks the nested_string and
+//     nested_int32 fields as filterable.
+//
+// EXPERIMENTAL: This option is experimental and may be changed or removed in the future.
+func WithFilterableFields(fields ...string) FilterOption {
+	return func(opts *filterOptions) {
+		opts.filterableFields = fields
+	}
+}
```

Comment:
> use the field name with a dot prefix

But the examples don't show a dot _prefix_ - they show dot annotation. Is that what you meant?